tests/integration/java_export/check-deploy-env.sh

```diff
@@ -0,0 +1,22 @@
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$0.runfiles/$f" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+  { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
+
+set -euox pipefail
+
+classes_file=$(rlocation rules_jvm_external/tests/integration/java_export/classes.txt)
+
+if grep -q Dependency.class "$classes_file"; then
+  exit 1
+fi
+
+if ! grep -q Main.class "$classes_file"; then
+  exit 2
```

Comment:
Why are the exit codes different?

Reply:
To differentiate why the test failed. Echoing a message instead.